Use supervisor on start and getting upstart to run on start event

taliesins opened the pull request:
- development now starts supervisor
- upstart successfully starts node.js on vagrant reload

taliesins committed:
- change box to lucid32
- development now starts supervisor
- upstart successfully starts node.js on vagrant reload

taliesins committed:
- ensure that node user can write to log file
- add logging for development

taliesins committed:
- When using development mode then access server directly
- When using production mode then access server via nginx
- Set path to watch for Supervisor
- Set path to node for Supervisor
- Ensure that logged in user owns log file for production

taliesins committed:
- Wait for local and remote file systems to be up before starting node.js server as we are storing file to serve in nfs share

taliesins committed:
- Web server port for production is always 80 as we want to go through nginx
- Web server port for development should connect directly to node.js

taliesins committed:
- Using maverick 64 as base box now as it has better support upstart events

taliesins committed:
- Add recipe to build redis from source

repeatingbeats commented:
Is the maverick 64 base box hosted publicly? I only want to pull that part in if I can document where people should download it. Everything else looks great and I'll get it merged as soon as I do some testing.

Thanks!

taliesins committed:
- Update to latest version of redis

taliesins committed:
- Add recipe to install erlang from source

taliesins committed:
- recipes for redis to install it from source added

taliesins committed:
- We might have multiple upstart templates so lets rename it to be specific for node.js

taliesins committed:
Pass in erlang version to use
Set username and password for riak

taliesins committed:
- Added a recipe to point to a shared folder for apt-get. Should help to alleviate some of the download pains.

taliesins committed:
- Include apt-get for production

taliesins committed:
- Make it possible to isolate each of the installed services in their own directories

taliesins committed:
- nodejs and npm installed into /usr/local
- move nodejs application creation out (going into its own recipe)

taliesins committed:
- Use 11.4 ubuntu server base box
- Add recipe to add nodejs application as a service

taliesins committed:
- User server.js instead of app.js for default application name

taliesins committed:
- User needs to take ownership of application directory

taliesins committed:
- Add rabbitmq

taliesins committed:
-create a nodejs-application recipe

taliesins committed:
-Must call nodejs application

taliesins committed:
Add example nodebox.json for default configuration

taliesins committed:
-Add ability to install packages as needed

taliesins committed:
-We want to ensure that symlink is setup before doing any apt-gets
-Provide a configurable list of packages to download and install

taliesins committed:
Add link to vagrant base boxes to use.

taliesins committed:
Create archive folder to store cached packages

taliesins committed:
Add riaksearch
Updated to 0.14.2-1 riak
Add link to nodebox.json from /var/www/web
Can specify packages via nodebox.json

taliesins committed:
Install riak from binary
Install riak-search from binary

taliesins committed:
Add riaksearch as a service
Generate configuration files for riaksearch
Specify configuration paths for riak and riaksearch

taliesins committed:
Update node.js tp 4.9
Update npm to 1.0.15
Fixed metadata for riak to point to correct version

taliesins committed:
Add cluster to default npm modules
NPM 1.0 changed the way you get modules and install them

taliesins committed:
Start rabbitmq server

taliesins committed:
Update node.js and npm
Open ports for debugging and riak browser

taliesins committed:
Remove RabbitMQ in favour of 0mq

taliesins committed:
Detect if 0mq is installed earlier so we don't have to download it all the time
